### PR TITLE
Milestone2

### PR DESCRIPTION
**Pull request by ioilmio**
This is the second milestone for the bookstore project it: 
- create the project structure
- implement the following component
  - App 
  - Booklist
  - BookForm
  - Book
- Define actions and a bookReducers
- Initialize  Redux store

### CODE REVIEW

File: src/containers/BookForm.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  'Sci-Fi',
+];
+const makeoption = categories.map((item, index) => (
+  <option key={index.toString()}>
+    {item}
+  </option>
+));
+
+const BookForm = () => (
+  <div>
```
Comment (addod19) [inline src/containers/BookForm.js:19]: Kindly change `<div>` to `<form>`

**Review by addod19 — CHANGES_REQUESTED** [review]
## Status: Require changes :recycle: :traffic_light: 

Hi @ioilmio, great job on this project :fireworks: The src/index.js file, App component, and the Book component is set up as per the requirements. However, there are a few issues we need to address before proceeding :+1: Kindly check the comments below :
- `BookList component` - We should use "mapStateToProps" to fetch books from the store
- `BookForm` component - Should render `<form>`, it is currently rendering `<div>` instead

Once these are addressed, we can submit another code review request :smile: 
_Happy coding_ :man_technologist: 
[Daniel Larbi Addo](https://www.linkedin.com/in/daniel-larbi-addo/)

File: src/containers/BookForm.js (hunk 1 above)
Comment (ioilmio) [reply src/containers/BookForm.js:19]: These changes are referred to the next milestone, this pull request is for milestone 2

**Comment by sumancrest0001** [comment]
### Status: Requires changes :gear: 

**Hello @ioilmio**, the project is running in the localhost, and most of the requirements from the milestone are implemented. There are a couple of things that need to be considered before moving ahead. Let's have a look at them.

**[Required]** Commit messages should be in the present tense and capitalized.

File: src/actions/index.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+export const CREATE_BOOK = ({ book }, state) => [
+  ...state,
+  book,
+];
+
+export const REMOVE_BOOK = ({ book }, state) => state.filter(
+  ({ bookId }) => bookId !== book.bookId,
+);
```
Comment (sumancrest0001) [inline src/actions/index.js]: **[Required]** These are the actions that need to be dispatched. Passed arguments don't match with what you are getting in the reducer. For example, in the CREATE_BOOK, you are returning an array but in the reducer, you are expecting a type and book as arguments. Keep it simple, just return an object with type and payload.

File: src/index.js [hunk 3]
```diff
@@ -1,17 +1,48 @@
...
 import ReactDOM from 'react-dom';
-import './index.css';
-import App from './App';
-import reportWebVitals from './reportWebVitals';
+import { Provider } from 'react-redux';
+import { createStore } from 'redux';
+import { v4 as uuidv4 } from 'uuid';
+import App from './components/App';
+import bookReducers from './reducers/index';
+
```
Comment (sumancrest0001) [inline src/index.js:8]: **[Optional]** You are not importing the bookReducer but the combineReducer. Try to use descriptive naming.

File: src/reducers/books.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
+const intitialState = [];
+
+const CREATE_BOOK = 'CREATE_BOOK';
+const REMOVE_BOOK = 'REMOVE_BOOK';
+
```
Comment (sumancrest0001) [inline src/reducers/books.js]: **[Required]** These are action types. We place it in the actions directory. Either place them inside the index.js file or create a new one for types only.

**Review by sumancrest0001 — CHANGES_REQUESTED** [review]
**Please make the requested changes and submit it for another code review.
If you have any queries or confusions regarding this code review, message me on Slack.
Suman Shrestha**

**Happy weekend !!!**

**Review by Torres-ssf — APPROVED** [review]

File: src/actions/index.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+// export const CREATE_BOOK = ({ book }, state) => [
+//   ...state,
+//   book,
+// ];
```
Comment (Torres-ssf) [inline src/actions/index.js]: - [ ] Please remove this commented code before merging the branch

**Review by Torres-ssf — APPROVED** [review]
### Milestone Approved :+1: 
Hi @ioilmio 
Awesome job implementing the requested changes
Good luck with the next milestone.

- [ ] **PLEASE check my comment bellow before merging the branch**

Happy coding :computer: 
**Sergio Torres**

File: src/actions/index.js (hunk 5 above)
Comment (ioilmio) [reply src/actions/index.js]: Done!
Thank you Sergio!

**Pull request merged**